sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java

```diff
@@ -37,6 +42,8 @@
      */
     private static final String DEFAULT_BUFFER_DIR = "sentry-buffered-events";
 
+    private static volatile ANRWatchDog anrWatchDog;
```

Comment:
What is the reason for this to be `volatile`? Are you enforcing a singleton pattern here?

Reply:
My thinking was: This is a static field in a non-static class. Although the documented way is to create a single factory when calling `Sentry.init`, the SDK could be initialized more than once, I didn't want to overkill and synchonize things there so the idea is to make sure the value of `anrWatchDog` is read from memory to reduce the chance of a race.

Reply:
I see what you mean.